src/components/app/content/cardList/card/index.js

```diff
@@ -0,0 +1,54 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import './index.css'
+import CardHeader from "./cardHeader";
+import CardBody from "./cardBody";
+
+function Card(props) {
+    const cardData = props.cardData;
+    const cardId = cardData.id;
+
+    const [tempState, setTempState] = useState({header: props.cardData.header, body: props.cardData.body});
+    const [editable, setEditable] = useState(false);
+
+    const cancelEditing = () => setEditable(false);
+
+    const saveCardDataChanges = () => {
+        cancelEditing();
+        props.saveCardData(cardId, tempState);
+    };
+    const restoreCardDataChanges = useCallback(() => {
+        cancelEditing();
+        setTempState({header: cardData.header, body: cardData.body});
+    }, [cardData.header, cardData.body]);
+
+    useEffect(() => {
+        restoreCardDataChanges();
+    }, [props.readOnly, restoreCardDataChanges]);
+
+    const fillHeader = (event) => setTempState({...tempState, header: event.target.value});
+    const fillBody = (event) => setTempState({...tempState, body: event.target.value});
```

Comment:
Again, these event handlers can be combined into one, if pass property name to update and set it as dynamic property.